### DISCUSSION TITLE
Command services registered as singletons?

### DISCUSSION THREAD

**Discussion by fbjerggaard** [Eventuous/eventuous#190] [General]
Hi,

I am working on a project where I "need" to inject some scoped services into the CommandService to access data saved earlier in the scope of the request.

That's where I spotted that the CommandService is registered as a singleton - are there any specific reasoning behind this? 

I tried registering it manually as a scoped service, and so far everything seems to be working as normal - using the postgres store and mongodb projections.

Just curious as to the design choice behind this, if there has been any deeper thoughts on the matter.

**Comment by alexeyzimarev** [comment]
It's not required to register command services as singletons. I made it a default as I rarely (or never) use transient or scoped dependencies there, and when I do I prefer using factories to control the scope inside the service rather than outside.

If you have to, there's nothing in the command service that prevents it from being registered as scoped or transient.

**Reply by fbjerggaard** [reply]
Perfect - I was just curious if there were any deeper reasoning behind it. I'll handle the registrations manually then.

Maybe an improvement idea would be to make the lifetime user-defined when calling `AddCommandService` - that would make it clear that there are nothing wrong doing it either way

**Reply by alexeyzimarev** [reply]
Yeah, I thought of the same thing.
